polygon/bridge/mdbx_store.go

```diff
@@ -292,30 +219,261 @@ func LastEventIDWithinWindow(tx kv.Tx, fromID uint64, toTime time.Time) (uint64,
 		}
 
 		if !event.Time.Before(toTime) {
-			return eventID, nil
+			return eventId, nil
 		}
 
-		eventID = event.ID
+		eventId = event.ID
 	}
 
-	return eventID, nil
+	return eventId, nil
 }
 
-func (s *MdbxStore) PutEvents(ctx context.Context, events []*heimdall.EventRecordWithTime) error {
+func (s *mdbxStore) PutEvents(ctx context.Context, events []*heimdall.EventRecordWithTime) error {
 	tx, err := s.db.BeginRw(ctx)
 	if err != nil {
 		return err
 	}
 	defer tx.Rollback()
 
-	if err = PutEvents(tx, events); err != nil {
+	if err = (txStore{tx}).PutEvents(ctx, events); err != nil {
 		return err
 	}
 
 	return tx.Commit()
 }
 
-func PutEvents(tx kv.RwTx, events []*heimdall.EventRecordWithTime) error {
+// Events gets raw events, start inclusive, end exclusive
+func (s *mdbxStore) Events(ctx context.Context, start, end uint64) ([][]byte, error) {
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	return txStore{tx}.Events(ctx, start, end)
+}
+
+func (s *mdbxStore) PutBlockNumToEventId(ctx context.Context, blockNumToEventId map[uint64]uint64) error {
+	if len(blockNumToEventId) == 0 {
+		return nil
+	}
+
+	tx, err := s.db.BeginRw(ctx)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	if err = (txStore{tx}).PutBlockNumToEventId(ctx, blockNumToEventId); err != nil {
+		return err
+	}
+
+	return tx.Commit()
+}
+
+// BlockEventIdsRange returns the [start, end] event Id for the given block number
+// ErrEventIdRangeNotFound is thrown if the block number is not found in the database.
+// If the given block number is the first in the database, then the first uint64 (representing start Id) is 0.
+func (s *mdbxStore) BlockEventIdsRange(ctx context.Context, blockNum uint64) (uint64, uint64, error) {
+	return s.blockEventIdsRange(ctx, blockNum, 9)
+}
+
+func (s *mdbxStore) blockEventIdsRange(ctx context.Context, blockNum uint64, lastFrozenId uint64) (uint64, uint64, error) {
+	var start, end uint64
+
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return start, end, err
+	}
+	defer tx.Rollback()
+
+	return txStore{tx}.blockEventIdsRange(ctx, blockNum, lastFrozenId)
+}
+
+func (s *mdbxStore) Unwind(ctx context.Context, blockNum uint64) error {
+	//
+	// TODO rename func to Unwind, unwind BorEventProcessedBlocks, BorTxnLookup - in separate PR
```

Comment:
can remove this - probably left over from a merge - it is all done & no longer needed

Reply:
I think this needs a bit more discussion - I think that UnwindEvents should not be necessary either - but I'm not sure how that relates to UnwindBlockNumToEventID & UnwindEventProcessedBlocks - and bridge.Unwind.